lib/enmeshed/connector_api.rb

```diff
@@ -0,0 +1,244 @@
+# frozen_string_literal: true
+
+module Enmeshed
+  class ConnectorError < StandardError; end
+
+  class ConnectorApi
+    # The app does not allow users to scan expired templates.
+    # However, previously scanned and then expired templates can still be submitted,
+    # resulting in the app silently doing nothing. CodeHarbor would still accept Relationships for expired templates if sent by the app.
+    # To minimize the risk of a template expiring before submission, we set the validity to 12 hours.
+    TEMPLATE_VALIDITY_PERIOD = 12.hours
+    # These attributes are mandatory in the app and must be provided.
+    # See https://enmeshed.eu/integrate/attribute-values for more attributes.
+    REQUIRED_ATTRIBUTES = %w[GivenName Surname EMailAddress].freeze
+
+    API_KEY = Settings.omniauth&.nbp&.enmeshed&.connector_api_key
+    CONNECTOR_URL = Settings.omniauth&.nbp&.enmeshed&.connector_url
+    DISPLAY_NAME = Settings.omniauth&.nbp&.enmeshed&.display_name
+
+    @conn = nil
+    @enmeshed_address = nil
+    @display_name_id = nil
+
+    def self.create_relationship_template(nbp_uid)
+      new_template = parse_result(conn.post('/api/v2/RelationshipTemplates/Own') do |req|
+        req.body = relationship_template(nbp_uid).to_json
+      end)
+
+      Rails.logger.debug { "Enmeshed::ConnectorApi RelationshipTemplate created: #{new_template[:truncatedReference]}" }
+      new_template[:truncatedReference]
+    end
+
+    def self.pending_relationship_for_nbp_uid(nbp_uid)
+      relationships = parse_relationships(conn.get('/api/v2/Relationships') do |req|
+        req.params['status'] = 'Pending'
+      end)
+
+      relationships = relationships.select do |rel|
+        # templates can only be scanned in their validity period but can theoretically be submitted infinitely late so we sanitize here
+        if rel.template_expiration < (TEMPLATE_VALIDITY_PERIOD * 2).ago
+          rel.reject
```

Comment:
A possible error is not rescued but bubbled up. It probably makes sense to comment this somewhere, maybe in the PORO `accept`/`reject` methods.

Reply:
@Dome-GER Do you have any further remarks on the objects we use or can we resolve this thread?

Reply:
Better now ;)